src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -237,16 +237,28 @@ void BaseCouplingScheme::initializeWithZeroInitialData(const DataMap &receiveDat
 
 PtrCouplingData BaseCouplingScheme::addCouplingData(const mesh::PtrData &data, mesh::PtrMesh mesh, bool requiresInitialization, bool communicateSubsteps, CouplingData::Direction direction)
 {
-  int             id = data->getID();
-  PtrCouplingData ptrCplData;
-  if (!utils::contained(id, _allData)) { // data is not used by this coupling scheme yet, create new CouplingData
```

Comment:
Do we still need `utils::contained`? Using `count` is definitely the better way to go. (no need to clean this up in this PR)

Reply:
[One day](https://en.cppreference.com/w/cpp/container/map/contains) :wink: